src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs

```diff
@@ -84,33 +92,66 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
 
-                if (declarationIndex > 0 &&
-                    sourceText.AreOnSameLine(block.Statements[declarationIndex - 1].GetLastToken(), localDeclarationStatement.GetFirstToken()))
+                bool TryFindPriorSurvivingStatementOnSameLine(out StatementSyntax statement)
+                {
+                    statement = null;
+
+                    var localDeclarationToken = localDeclarationStatement.GetFirstToken();
+                    for (int i = declarationIndex - 1; i >= 0; i--)
+                    {
+                        var statementSyntax = block.Statements[i];
+                        if (declarationsToRemove.Contains(statementSyntax))
+                        {
+                            continue;
+                        }
+                        if (sourceText.AreOnSameLine(statementSyntax.GetLastToken(), localDeclarationToken))
+                        {
+                            statement = statementSyntax;
+                            return true;
+                        }
+                    }
+                    return false;
+                }
+
+                if (declarationIndex > 0 && TryFindPriorSurvivingStatementOnSameLine(out var priorStatementSyntax))
                 {
                     // There's another statement on the same line as this declaration statement.
                     // i.e.   int a; int b;
                     //
                     // Just move all trivia from our statement to be trailing trivia of the previous
                     // statement
                     editor.ReplaceNode(
-                        block.Statements[declarationIndex - 1],
+                        priorStatementSyntax,
                         (s, g) => s.WithAppendedTrailingTrivia(localDeclarationStatement.GetTrailingTrivia()));
                 }
                 else
                 {
                     // Trivia on the local declaration will move to the next statement.
-                    // use the callback form as the next statement may be the place where we're 
+                    // use the callback form as the next statement may be the place where we're
                     // inlining the declaration, and thus need to see the effects of that change.
+
+                    // Find the next Statement that isn't going to be removed
+                    StatementSyntax nextStatementSyntax = null;
+                    for (int i = declarationIndex + 1; i < block.Statements.Count; i++)
+                    {
+                        var statement = block.Statements[i];
+                        if (!declarationsToRemove.Contains(statement))
+                        {
+                            nextStatementSyntax = statement;
+                            break;
+                        }
+                    }
+
                     editor.ReplaceNode(
-                        block.Statements[declarationIndex + 1],
+                        nextStatementSyntax,
```

Comment:
this coudl end up being null.

Reply:
can you add a comment to that effect.  THanks!

Reply:
Done

Reply:
@Suchiman Wouldn't an assertion be more appropriate?